### PR TITLE
D10 compatible version of the package

### PR DESCRIPTION
**Pull request by vermario**
> ATTENTION! The changes in this branch mean that the package will not work with drupal9.

This PR is against the new 10.x branch.

This branch adds Drupal 10 support to the package, by

* upgrading to codeception5 (because only this version supports symfony 6 which drupal10 uses
* upgrading to module-webdriver 3 (this still supports phpunit9, version 4 requires phpunit10)
* adding small fixes to the existing integration code to make d10 happy (details in comments)

## To test

1.  add this repository to your project's composer.json `repositories` section:

```
  {
            "type": "vcs",
            "url": "https://github.com/vermario/drupal-codeception"
        },
``` 

2. `composer require wunderio/drupal-codeception:10.x --dev`

The tests should run as usual. 

I have not tried this with drupal9, but in theory it should work there as well. Could you help testing it on drupal 9? 

if it works, then maybe we can merge this over the 9.x branch? or how do we create a new version in composer?

### CODE REVIEW

File: src/Codeception/TestDrupalKernel.php [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 use Drupal\Core\DrupalKernel;
 use Drupal\Core\Site\Settings;
-use Symfony\Cmf\Component\Routing\RouteObjectInterface;
+use Drupal\Core\Routing\RouteObjectInterface;
```
Comment (vermario) [inline src/Codeception/TestDrupalKernel.php:7]: No more  `Symfony\Cmf\Component\Routing\RouteObjectInterface` , change record:  https://www.drupal.org/node/3151009

File: src/Codeception/Module/DrupalEntity.php [hunk 2]
```diff
@@ -5,7 +5,7 @@
 use Codeception\Module;
 use Drupal\Core\Entity\FieldableEntityInterface;
 use Drupal\Core\Url;
-use Codeception\TestCase;
```
Comment (vermario) [inline src/Codeception/Module/DrupalEntity.php:8]: This class no longer exists, switch to `TestInterface`

File: src/Codeception/Module/DrupalEntity.php [hunk 3]
```diff
@@ -30,7 +30,7 @@ class DrupalEntity extends Module {
    *
    * @var array
    */
-  protected $config = [
+  protected array $config = [
```
Comment (vermario) [inline src/Codeception/Module/DrupalEntity.php:33]: needs to be explicitly set to array to match the declaration

**Comment by Kopin1** [comment]
Hello! What is the status on this?

**Comment by dajjen** [comment]
I have tested this in Drupal 10.1 and it works good!
When to merge this PR?

**Comment by vermario** [comment]
Merging to the 10.x branch. Thanks for testing @dajjen

**Comment by dajjen** [comment]
Thank you @vermario

**Pull request merged**